app/enterprise_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package app
+
+import (
+	"testing"
+
+	"github.com/mattermost/mattermost-server/v5/einterfaces"
+	"github.com/mattermost/mattermost-server/v5/einterfaces/mocks"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEnterpriseNone(t *testing.T) {
+
+	th := SetupEnterprise(t).InitBasic()
+	defer th.TearDown()
+
+	th.App.UpdateConfig(func(cfg *model.Config) {
+		*cfg.ExperimentalSettings.UseNewSAMLLibrary = true
+	})
+
+	assert.Nil(t, th.App.Srv.Saml)
```

Comment:
This result is not obvious to me. Or put another way, I don't know what `TestEnterpriseNone` means. Should we have subtests instead with more verbose descriptions?